### ISSUE TITLE
Parse regular expressions in card contents filter & card search

### ISSUE THREAD

**Issue by sibir-ine** [loot/loot#1689]
I've split this off from #1189 as it's probably worthy of a separate issue.

An advanced filtering mode for the card contents filter would benefit users as it could display multiple related plugins that don't contain the same contents. Users could also use it when a phrase would display many plugins irrelevant to the user. Since the filter searches for any text within the card (not just plugin filenames), the best way of approaching this would likely be using regexes. We could add a toggle to enable a mode for inputting regexes. Then, any cards with text matching the regex would remain displayed.

We could also extend this to the card search feature for similar reasons & in a similar fashion.

**Comment by Ortham** [comment]
@sibir-ine I've implemented this in the plugin-sidebar-indexes branch, is it as you expect?

**Comment by sibir-ine** [comment]
Yup, the filter works well, thanks 👍 As a side note, it looks like you need to log in to access the snapshot builds from Artifactory, so I had to get it from Actions.

**Comment by Ortham** [comment]
> As a side note, it looks like you need to log in to access the snapshot builds from Artifactory, so I had to get it from Actions.

That's new... I'll look into it.

**Comment by Ortham** [comment]
Done in f8797d58960f4219e51624570f73a1a9397d58fe.

**Comment by sibir-ine** [comment]
Would it be feasible to add the feature to the search dialog too, or does it not seem worth the effort?

**Comment by Ortham** [comment]
> Would it be feasible to add the feature to the search dialog too, or does it not seem worth the effort?

I've added that in 5fc0f04be48408e129dbd413af170c10365f744c.
